src/DataFrames.jl

```diff
@@ -128,4 +128,18 @@ for (dir, filename) in [
     include(joinpath(dir, filename))
 end
 
+# Import  "StringEncodings" if available and set up decoder
+_decoder=nothing
+ENCODING_PACKAGE_READY = false
+function __init__()
+    try Pkg.installed("StringEncodings")
```

Comment:
Can you kindly explain why will it interact badly with pre-compilation? the `__init__` function block will not be precompiled, everything is else will. At least that is what  I think is happening. 
Also `StringEncdoings` will not be on the `LOAD_PATH`.